agent/stats/task_windows.go

```diff
@@ -17,64 +18,42 @@ package stats
 
 import (
 	"context"
-	"os/exec"
-	"strconv"
-	"strings"
 	"time"
 
 	"github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/eni/networkutils"
 	"github.com/aws/amazon-ecs-agent/agent/stats/resolver"
 
 	dockerstats "github.com/docker/docker/api/types"
 	"github.com/pkg/errors"
 )
 
-const (
-	receivedBroadcastPackets = "ReceivedBroadcastPackets"
-	receivedMulticastPackets = "ReceivedMulticastPackets"
-	receivedUnicastPackets   = "ReceivedUnicastPackets"
-	sentBroadcastPackets     = "SentBroadcastPackets"
-	sentMulticastPackets     = "SentMulticastPackets"
-	sentUnicastPackets       = "SentUnicastPackets"
-	receivedBytes            = "ReceivedBytes"
-	receivedPacketErrors     = "ReceivedPacketErrors"
-	receivedDiscardedPackets = "ReceivedDiscardedPackets"
-	sentBytes                = "SentBytes"
-	outboundPacketErrors     = "OutboundPacketErrors"
-	outboundDiscardedPackets = "OutboundDiscardedPackets"
-)
-
-var (
-	// Making it visible for unit testing
-	execCommand = exec.Command
-	// Fields to be extracted from the stats returned by cmdlet.
-	networkStatKeys = []string{
-		receivedBroadcastPackets,
-		receivedMulticastPackets,
-		receivedUnicastPackets,
-		sentBroadcastPackets,
-		sentMulticastPackets,
-		sentUnicastPackets,
-		receivedBytes,
-		receivedPacketErrors,
-		receivedDiscardedPackets,
-		sentBytes,
-		outboundDiscardedPackets,
-		outboundPacketErrors,
-	}
-)
-
 type StatsTask struct {
 	*statsTaskCommon
+	interfaceLUID []uint64
+	netUtils      networkutils.NetworkUtils
 }
 
 func newStatsTaskContainer(taskARN, taskId, containerPID string, numberOfContainers int,
 	resolver resolver.ContainerMetadataResolver, publishInterval time.Duration, taskENIs task.TaskENIs) (*StatsTask, error) {
 	ctx, cancel := context.WithCancel(context.Background())
 
+	// Instantiate an instance of network utils.
+	// This interface would be used to invoke Windows networking APIs.
+	netUtils := networkutils.New()
+
 	devices := make([]string, len(taskENIs))
+	ifaceLUID := make([]uint64, len(taskENIs))
+	// Find and store the device name along with the interface LUID.
 	for index, device := range taskENIs {
 		devices[index] = device.LinkName
+
+		ifEntry, err := netUtils.GetMIBIfEntryFromDeviceName(device.LinkName)
+		if err != nil {
+			cancel()
+			return nil, errors.Wrapf(err, "failed to initialise stats task container")
+		}
+		ifaceLUID[index] = ifEntry.InterfaceLUID
```

Comment:
I still dont quite understand. Cant we just use `ConvertInterfaceIndexToLuid` for this and then be done? We dont need to do an entire mib table and filter it right?

Reply:
So the thing is that we do not store interface index for the task network adapters. There can be two cases here-
- We query the interface index first using `net.Interfaces` and then use `ConvertInterfaceIndexToLuid` in the stats module itself. This leads to two Win32 calls. Now, `net.Interfaces` would in turn call `GetAdaptersAddresses` which fetches all the addresses associated with the adapters. I feel that a single call to `GetIfTable2Ex` might be more performant than these two calls.
- We store the ENI's interface index along with Link name during the initial task attachment workflow. Now, the issue here is that if the interface index becomes stale for some reason say, instance reboot, we would not have any mechanism to identify or refresh the same.